phoenix-core-client/src/main/java/org/apache/phoenix/iterate/OrderedResultIterator.java

```diff
@@ -172,6 +188,28 @@ public OrderedResultIterator(ResultIterator delegate,
         this(delegate, orderByExpressions, spoolingEnabled, thresholdBytes, limit, offset, estimatedRowSize, Long.MAX_VALUE);
     }
 
+    public OrderedResultIterator(ResultIterator delegate,
+                                 List<OrderByExpression> orderByExpressions,
+                                 boolean spoolingEnabled,
+                                 long thresholdBytes, Integer limit, Integer offset,
+                                 int estimatedRowSize, long pageSizeMs, Scan scan,
+                                 RegionInfo regionInfo) {
+        this(delegate, orderByExpressions, spoolingEnabled, thresholdBytes, limit, offset,
+                estimatedRowSize, pageSizeMs);
+        this.scan = scan;
+        // If scan start rowkey is empty, use region boundaries. Reverse region boundaries
+        // for reverse scan.
+        this.scanStartRowKey = scan.getStartRow().length > 0 ? scan.getStartRow() :
```

Comment:
I see this logic being used in many places. If we can add a util function for this too it will be helpful

Reply:
Can do this but now that we have server and client modules, we won't be able to use one util by all coprocs. Except for `OrderedResultIterator`, every other coproc can use it. This happens if we move util function to server util class.

On the other hand, we can't move util function to core client module because client module does not have access to hbase server classes e.g. `Region`.